src/core/public/application/application_service.tsx

```diff
@@ -209,7 +212,8 @@ export class ApplicationService {
         }
 
         const appBasePath = basePath.prepend(appRoute);
-        const mount: LegacyAppMounter = () => redirectTo(appBasePath);
+        const mount: LegacyAppMounter = ({ history: appHistory }) =>
+          redirectTo(appHistory.createHref(appHistory.location));
```

Comment:
I never was a big fan of the decision to perform the KP->LP page reload in the LP `mount` method. My feeling is that changing `start.navigateToApp` or `service.navigate` to perform the `redirect` instead of mounting the stub legacy app that will perform a redirect would probably be way safer and cleaner, wdyt? (of course it totally  breaks every IT tests for this behavior)

Reply:
Yes I think that is the only real solution, otherwise we really have two page navigations: the history.push followed by the location.assign.

Reply:
Realized I could move the `reload` call into the legacy mounter, tests passed locally now. Good because I wanted to avoid refactoring this if before we remove this altogther.